### PR TITLE
Add colors and erasors

### PR DESCRIPTION
**Pull request by Eczbek**
- `Color` structure
- Presets of basic colors
- `fg` function to set foreground
- `bg` function to set background
- `colors.cpp` example
- Erasors:
    - `clear_screen`
    - `clear_screen_until`
    - `clear_screen_from`
    - `clear_line`
    - `clear_line_until`
    - `clear_line_from`
- Removed `detail::die`
- Added and removed some inclusions

### CODE REVIEW

File: rawterm.h [hunk 1]
```diff
@@ -33,6 +33,8 @@
 #include <unistd.h>
 
 #include <algorithm>
+#include <cstddef>
+#include <cstdint>
```
[the remark this answers is not in the document]
Comment (Eczbek) [reply rawterm.h:37]: `<cstddef>` is for `std::size_t`
`<cstdint>` is for `std::uint8_t`

[the remark this answers is not in the document]
Comment (Ttibsi) [reply rawterm.h:37]: We shouldn't need `cstddef` as `cstdio` includes `std::size_t` already on the line below

**Comment by Ttibsi** [comment]
Can you add the example to build.sh as well please?

File: rawterm.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (Eczbek) [reply rawterm.h:37]: We shouldn't rely on that inclusion. Also there may be better alternatives to `perror`, which if replaced, can allow us to remove the inclusion of `<cstdio>`

File: examples/colors.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		rawterm::cyan,
+		rawterm::azure,
+		rawterm::blue,
+		rawterm::violet,
+		rawterm::purple,
+		rawterm::magenta
+	};
+
+	for (rawterm::Color color : colors) {
+		std::cout << rawterm::fg(rawterm::bold("Hello, world!\n"), color);
```
Comment (Ttibsi) [inline examples/colors.cpp:25]:
```suggestion
		std::cout << rawterm::fg(rawterm::bold("Hello, world!\r\n"), color);
```

File: examples/colors.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	}
+
+	for (rawterm::Color color : colors) {
+		std::cout << rawterm::bg(rawterm::bold("Hello, world!\n"), color);
```
Comment (Ttibsi) [inline examples/colors.cpp:29]:
```suggestion
		std::cout << rawterm::bg(rawterm::bold("Hello, world!\r\n"), color);
```

**Comment by Eczbek** [comment]
I think the `std::flush`es are unnecessary

File: rawterm.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (Eczbek) [reply rawterm.h:36]: [C++ exceptions replace `perror`](https://en.cppreference.com/w/cpp/header/stdexcept)

[the remark this answers is not in the document]
Comment (Ttibsi) [reply rawterm.h:36]: This seems to be fine, so we can make that change as well

**Comment by Ttibsi** [comment]
```diff
--- a/rawterm.h
+++ b/rawterm.h
@@ -137,9 +137,9 @@ namespace rawterm {
         static const std::unordered_set<char> asciiLetters{
             '\x31', '\x32', '\x33', '\x34', '\x35', '\x36', '\x37', '\x38',
             '\x39', '\x41', '\x42', '\x43', '\x44', '\x45', '\x46', '\x47',
-            '\x48', '\x49', '\x4A', '\x4B', '\x4C', '\x4D', '\x4E', /* '\x4F', */
+            '\x48', '\x49', '\x4A', '\x4B', '\x4C', '\x4D', '\x4E', '\x4F',
             '\x50', '\x51', '\x52', '\x53', '\x54', '\x55', '\x56', '\x57',
-            '\x58', '\x59', '\x5A', '\x61', '\x62', '\x63', '\x66', '\x67',
+            '\x58', '\x59', '\x5A', '\x61', '\x62', '\x63', '\x64', '\x65',
             '\x66', '\x67', '\x68', '\x69', '\x6A', '\x6B', '\x6C', '\x6D',
             '\x6E', '\x6F', '\x70', '\x71', '\x72', '\x73', '\x76', '\x77',
             '\x76', '\x77', '\x78', '\x79', '\x7A'};
```

Can we make the above change please - some alt_l+characters weren't working.

Note that the uses of `std::flush` were found to be needed because at least on `move_cursor()`, I've found that the cursors weren't moving without flushing those codes through. I'm not sure what exactly was causing it, but I needed that to work

**Comment by Ttibsi** [comment]
Git messing up here, had no intention to close this PR -- see #20

**Pull request closed without merging**